framework/src/userobject/InterfaceQpMaterialPropertyBaseUserObject.C

```diff
@@ -0,0 +1,13 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// MOOSE includes
+#include "InterfaceQpMaterialPropertyBaseUserObject.h"
+
+defineLegacyParams(InterfaceQpMaterialPropertyBaseUserObject<>);
```

Comment:
Should be able to remove this entire file

Reply:
ideally yes, but if I move everything to header then it sone't like it (compiles but complains about missing reference before execution)

Reply:
I meant that we didn't need any of the code in this file, e.g. you should delete the code...but if legacy really doesn't mean legacy then you're good